djvm/src/main/kotlin/net/corda/djvm/analysis/AnalysisConfiguration.kt

```diff
@@ -328,6 +341,51 @@ class AnalysisConfiguration private constructor(
                 }
             }.withBody()
              .build()
+        ).mapByClassName() + listOf(
+            object : MethodBuilder(
+                access = ACC_STATIC or ACC_PRIVATE,
+                className = sandboxed(Security::class.java),
+                memberName = "initialize",
+                descriptor = "()V"
+            ) {
+                override fun writeBody(emitter: EmitterModule) = with(emitter) {
+                    invokeStatic("sandbox/java/lang/DJVM", "getSecurityProviders", "()Lsandbox/java/util/Properties;")
+                    putStatic(className, "props", "Lsandbox/java/util/Properties;")
+                    returnVoid()
+                }
+            }.withBody()
+             .build()
+        ).mapByClassName() + listOf(
+            // This method will be deleted.
+            Member(
+                access = ACC_STATIC,
+                className = sandboxed(Modifier::class.java),
+                memberName = "<clinit>",
+                descriptor = "()V",
+                genericsDetails = ""
+            )
+        ).mapByClassName() + listOf(
+            // This method will be deleted.
+            Member(
+                access = ACC_STATIC,
+                className = sandboxed(Random::class.java),
+                memberName = "<clinit>",
+                descriptor = "()V",
+                genericsDetails = ""
+            )
+        ).mapByClassName() + listOf(
+            object : MethodBuilder(
+                access = ACC_PRIVATE or ACC_STATIC,
+                className = sandboxed(SecureRandom::class.java),
+                memberName = "getPrngAlgorithm",
+                descriptor = "()Lsandbox/java/lang/String;"
+            ) {
+                override fun writeBody(emitter: EmitterModule) = with(emitter) {
+                    pushNull()
+                    returnObject()
+                }
+            }.withBody()
+             .build()
```

Comment:
Are these STICHED_CLASSES some ad-hoc tweaks that you make to the original classes so that they work in the sandbox?

Reply:
Yes. Originally it was adding small methods that the compiler would have added automatically anyway, such as bridge methods. But now it also allows me to replace / delete certain methods on classes to save me from reimplementing those classes from scratch.

It is important to use as much of the original byte-code as possible; i don't want the number of template classes to snowball out of control.